meteor/lib/api/rest/v1/buckets.ts

```diff
@@ -0,0 +1,124 @@
+import { Meteor } from 'meteor/meteor'
+import { ClientAPI } from '../../client'
+import {
+	BucketAdLibActionId,
+	BucketAdLibId,
+	BucketId,
+	ShowStyleBaseId,
+} from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { IngestAdlib } from '@sofie-automation/blueprints-integration'
+
+export interface BucketsRestAPI {
+	/**
+	 * Get all available Buckets.
+	 *
+	 * @param connection Connection data including client and header details
+	 * @param event User event string
+	 * @param inputs Migration data to apply
+	 */
+	getAllBuckets(
+		connection: Meteor.Connection,
+		event: string
+	): Promise<ClientAPI.ClientResponse<Array<APIBucketComplete>>>
+
+	/**
+	 * Get a Bucket.
+	 *
+	 * @param connection Connection data including client and header details
+	 * @param event User event string
+	 * @param inputs Migration data to apply
+	 */
+	getBucket(
+		connection: Meteor.Connection,
+		event: string,
+		bucketId: BucketId
+	): Promise<ClientAPI.ClientResponse<APIBucketComplete>>
+
+	/**
+	 * Adds a new Bucket, returns the Id of the newly created Bucket.
+	 *
+	 * @param connection Connection data including client and header details
+	 * @param event User event string
+	 * @param bucket Bucket to add
+	 */
+	addBucket(
+		connection: Meteor.Connection,
+		event: string,
+		bucket: APIBucket
+	): Promise<ClientAPI.ClientResponse<BucketId>>
+
+	/**
+	 * Deletes a Bucket.
+	 *
+	 * @param connection Connection data including client and header details
+	 * @param event User event string
+	 * @param bucketId Id of the bucket to delete
+	 */
+	deleteBucket(
+		connection: Meteor.Connection,
+		event: string,
+		bucketId: BucketId
+	): Promise<ClientAPI.ClientResponse<void>>
+
+	/**
+	 * Empties a Bucket.
+	 *
+	 * @param connection Connection data including client and header details
+	 * @param event User event string
+	 * @param bucketId Id of the bucket to empty
+	 */
+	emptyBucket(
+		connection: Meteor.Connection,
+		event: string,
+		bucketId: BucketId
+	): Promise<ClientAPI.ClientResponse<void>>
+
+	/**
+	 * Deletes a Bucket AdLib.
+	 *
+	 * @param connection Connection data including client and header details
+	 * @param event User event string
+	 * @param adLibId Id of the bucket adlib to delete
+	 */
+	deleteBucketAdLib(
+		connection: Meteor.Connection,
+		event: string,
+		adLibId: BucketAdLibId | BucketAdLibActionId
```

Comment:
I wonder if this is the correct id to use, of if it should be the `externalId`.

Internally, when the import happens it can generate one or more bucket adlibs, one for each ShowStyleVariant. We 'group' them by the externalId when necessary.  
Perhaps this api should do the same?

This is nothing like how the operation is done today, but the current implementation is not great in this regard. It would be nice to not bake this limitation/quirk into the stable api too

Reply:
This has been updated to allow deleting by externalId